File: src/main/java/com/github/klaidoshka/vehiclecrashes/constant/VehicleType.java
package com.github.klaidoshka.vehiclecrashes.constant;

import org.springframework.lang.NonNull;

public enum VehicleType {

  CAR(0),
  BUS(1),
  VAN(2),
  TRUCK(3),
  MOTORCYCLE(4),
  BICYCLE(5),
  BOAT(6),
  PLANE(7);

  private final int id;

  VehicleType(int id) {
    this.id = id;
  }

  public static @NonNull VehicleType fromId(@NonNull int vehicleTypeId) {
    for (VehicleType type : values()) {
      if (type.getId() == vehicleTypeId) {
        return type;
      }
    }

    throw new IllegalArgumentException("No vehicle type with id " + vehicleTypeId);
  }

  public int getId() {
    return id;
  }

  @Override
  public String toString() {
    return name();
  }
}

File: client/src/constants/VehicleType.ts
enum VehicleType {
  CAR = 0,
  BUS = 1,
  VAN = 2,
  TRUCK = 3,
  MOTORCYCLE = 4,
  BICYCLE = 5,
  BOAT = 6,
  PLANE = 7
}

const values = Object
  .keys(VehicleType)
  .filter((key: string | number) => isNaN(key as number))
  .map((key: string | number) => key as number);

export {VehicleType, values};